packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -788,13 +788,38 @@ class _CupertinoBackGestureController<T> {
     }
 
     if (controller.isAnimating) {
+      late VoidCallback animationValueCallback;
+      late AnimationStatusListener animationStatusCallback;
+
+      // Keep the userGestureInProgress in true state so we don't change the
+      // curve of the page transition mid-flight since CupertinoPageTransition
+      // depends on userGestureInProgress. Switch the userGestureInProgress to
+      // false near the end, but just before so the app can be interacted with
+      // when the animation visibly ends.
+      animationValueCallback = () {
```

Comment:
If we could make `CupertinoPageTransition`'s animation curve to be resilient between user drag and ballistic animation. We can just call navigator.didStopUserGesture(); directly without this 0.02 workaround right? It seems to me that will be a better solution if we can make it so. One way to do it is make CupertinoPageTransition a statefulwidget and only update the _primaryPositionAnimation and _secondaryPositionAnimation when the input animation status changes. WDYT?

Reply:
As in on drag end, we'd update _primaryPositionAnimation? `navigator.didStopUserGesture()` isn't called elsewhere and I'm not sure where we'd move it to in that case. By calling it directly do you mean differently from the status listener that already existed?

Reply:
No I think the reason we can't call navigator.didStopUserGesture() is that it will change cause `CupertinoRouteTransitionMixin.isPopGestureInProgress` to return false. `CupertinoPageTransition` to switch its _primaryPositionAnimation/_secondaryPositionAnimation from linear transition to CurvedAnimation thus may result in a jump. See https://github.com/flutter/flutter/blob/bebea0cc6c13da7673f597d71616f443ed4672d9/packages/flutter/lib/src/cupertino/route.dart#L472

If we can make it so that CupertinoPageTransition will not switch the animation when CupertinoRouteTransitionMixin.isPopGestureInProgress change from true to false if the animation is ongoing, we can directly call `navigator.didStopUserGesture()` here which will unblock user from tapping the screen.

Reply:
Ok, that makes sense. That explains some things I was seeing in the tests.